database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java

```diff
@@ -269,15 +278,16 @@ private void setResultSetMetadata() throws Exception {
     return columnTypes;
   }
 
-  private void validateSchema(Class<? extends Driver> jdbcDriverClass, String tableName, Schema inputSchema) {
+  private void validateSchema(FailureCollector collector, Class<? extends Driver> jdbcDriverClass, String tableName,
+                              Schema inputSchema) {
     String connectionString = dbSinkConfig.getConnectionString();
 
     try {
       DBUtils.ensureJDBCDriverIsAvailable(jdbcDriverClass, connectionString, dbSinkConfig.jdbcPluginName);
     } catch (IllegalAccessException | InstantiationException | SQLException e) {
-      throw new InvalidStageException(String.format("Unable to load or register JDBC driver '%s' while checking for " +
-                                                      "the existence of the database table '%s'.",
-                                                    jdbcDriverClass, tableName), e);
+      collector.addFailure(String.format("Unable to load or register JDBC driver '%s' while checking for " +
+                                           "the existence of the database table '%s'.",
+                                         jdbcDriverClass, tableName), null).withStacktrace(e.getStackTrace());
```

Comment:
if this fails, doesn't that mean the other logic will fail?

Reply:
yea, thats true. Modified the code to incorporate this.